src/nupic/frameworks/opf/metrics.py

```diff
@@ -245,119 +319,116 @@ def _isNumber(value):
 
 class MetricsIface(object):
   """
-  A Metrics module compares a prediction Y to corresponding ground truth X and returns a single
-  measure representing the "goodness" of the prediction. It is up to the implementation to
-  determine how this comparison is made.
+  A Metrics module compares a prediction Y to corresponding ground truth X and 
+  returns a single measure representing the "goodness" of the prediction. It is 
+  up to the implementation to determine how this comparison is made.
 
+  :param metricSpec: (:class:`MetricSpec`) spec used to created the metric
   """
 
   __metaclass__ = ABCMeta
 
   @abstractmethod
   def __init__(self, metricSpec):
-    """
-        instantiate a MetricsIface-based module.
-        args:
-            metricSpec is an instance of MetricSpec
-
-    """
+    pass
 
   @abstractmethod
   def addInstance(self, groundTruth, prediction, record = None, result = None):
-    """ add one instance consisting of ground truth and a prediction.
-
-      Parameters:
-      -----------------------------------------------------------------------
-      groundTruth:
-        The actual measured value at the current timestep
-      prediction:
-        The value predicted by the network at the current timestep
-
-      groundTruthEncoding:
-        The binary encoding of the groundTruth value (as a numpy array). Right
-        now this is only used by CLA networks
-      predictionEncoding:
-        The binary encoding of the prediction value (as a numpy array). Right
-        now this is only used by CLA networks
-
-      result:
-        An ModelResult class (see opf_utils.py)
-
-        return:
-            The average error as computed over the metric's window size
+    """ 
+    Add one instance consisting of ground truth and a prediction.
+
+    :param groundTruth:
+      The actual measured value at the current timestep
+    
+    :param prediction:
+      The value predicted by the network at the current timestep
+
+    :param record: ?
+
+    :param result: (:class:`nupic.frameworks.opf.opf_utils.ModelResult`) the
+           result of running a row of data through an OPF model
+
+    :returns:
+        The average error as computed over the metric's window size
     """
 
   @abstractmethod
   def getMetric(self):
     """
-        return:
-            {value : <current measurement>, "stats" : {<stat> : <value> ...}}
-            metric name is defined by the MetricIface implementation. stats is expected to contain further
-                information relevant to the given metric, for example the number of timesteps represented in
-                the current measurement. all stats are implementation defined, and "stats" can be None
-
+    Metric name is defined by the MetricIface implementation. ``stats`` is 
```

Comment:
I have the same confusion as you. What is metric name? It looks safe to me to remove any reference to a metric name since I don't think there is one that is relevant to this function.

The return value is also unclear since you say it is an object but what kind? Is it an instance of a specific class or a dict? This is important because it determines whether you do `returnValue.value`, `returnValue["value"]`, or `returnValue.value()` to get the value form the returned object.